src/devices/tuya.ts

```diff
@@ -10968,6 +10968,50 @@ const definitions: Definition[] = [
             ],
         },
     },
+    {
+        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE204_ncti2pro'}],
+        model: 'TS0601_pn6_6_way_controller',
+        vendor: 'Tuya',
+        description: '6-way controller',
+        whiteLabel: [tuya.whitelabel('ZSVIOT', 'PN6', '6-way controller', ['_TZE204_ncti2pro'])],
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            tuya.exposes.switch(),
+            tuya.exposes.switchMode2().withEndpoint('l1_l2').withDescription('1-2 channels'),
```

Comment:
Could you use `e.binary()` for these instead?

Reply:
When I use `e.binary('switch_mode', ea.STATE_SET, 'switch', 'curtain')`, the switch control appears as a single line of switches, which I believe is not the best UI solution:

![Screenshot 2024-08-08 at 23 51 41](https://github.com/user-attachments/assets/5d63de50-424d-43ac-9dbf-1cbc90aa053d)

Using `e.enum` wrapped into `switchMode2` presents it as a segmented control, which I find more appropriate than a simple switch:

![Screenshot 2024-08-09 at 00 16 43](https://github.com/user-attachments/assets/26279ae7-e574-4ee9-b8ff-1e58acb098f2)

Here is the original UI from the Tuya app for reference:

![IMG_EF090012F0F8-1](https://github.com/user-attachments/assets/1a34c78f-fd21-406c-8eef-ad364dbaa072)

Additionally, I created a new expose function and value converter to avoid duplicating code:

```typescript
exposes: [
    tuya.exposes.switch(),
    e.enum('switch_mode', ea.STATE_SET, ['switch', 'curtain'])
        .withDescription('Sets the mode of the switch to act as a switch or as a curtain controller')
        .withEndpoint('l1_l2')
        .withLabel('1-2 channels'),
    tuya.exposes.switch().withEndpoint('l1'),
    tuya.exposes.switch().withEndpoint('l2'),
    e.enum('switch_mode', ea.STATE_SET, ['switch', 'curtain'])
        .withDescription('Sets the mode of the switch to act as a switch or as a curtain controller')
        .withEndpoint('l3_l4')
        .withLabel('3-4 channels'),
    tuya.exposes.switch().withEndpoint('l3'),
    tuya.exposes.switch().withEndpoint('l4'),
    e.enum('switch_mode', ea.STATE_SET, ['switch', 'curtain'])
        .withDescription('Sets the mode of the switch to act as a switch or as a curtain controller')
        .withEndpoint('l5_l6')
        .withLabel('5-6 channels'),
    tuya.exposes.switch().withEndpoint('l5'),
    tuya.exposes.switch().withEndpoint('l6'),
    tuya.exposes.switchType(),
    e.power_on_behavior(['off', 'on']).withAccess(ea.STATE_SET),
]
```

However, if code duplication is preferred over creating a new expose function and value converter, I'm happy to revert to using the duplicated code.

Reply:
I see, this is fine. Thanks!